### PR TITLE
Adding creation/management for SASL/SCRAM SHA-512 Kafka Users

### PR DESCRIPTION
**Pull request by charlese-instaclustr**
Updates the terraform provider for compatibility with the updated Instaclustr Kafka user management API. 

The API has recently had support added for a new 'options' property, within which one can specify the `sasl-scram-mechanism` with which a user's password should be authenticated -- the options being `SCRAM-SHA-256` or `SCRAM-SHA-512`. This can be set upon user creation or user password reset.

This PR allows usage of this property via the terraform provider. It also maintains backward compatibility, such that old plans and resource descriptions involving Kafka users should not be negatively affected.

### CODE REVIEW

File: instaclustr/resource_kafka_user.go
Comment (EngineersBox) [inline instaclustr/resource_kafka_user.go:57]: **[Question]** I'm curious about the use of `ForceNew: true` for both the options schema and the SCRAM field. Is there a reason that both have to be re-created? My gut feeling is that it accounts for future options settings, but still seems overkill.

**Review by EngineersBox — CHANGES_REQUESTED** [review]
Looking good, some comments around TF schema and general go stuff

File: instaclustr/resource_kafka_user.go
Comment (charlese-instaclustr) [reply instaclustr/resource_kafka_user.go:56]: Addressed

File: acc_test/data/kafka_user_create_user.tf
Comment (charlese-instaclustr) [reply acc_test/data/kafka_user_create_user.tf:50]: Good option, working this in now

Comment (charlese-instaclustr) [reply acc_test/data/kafka_user_create_user.tf:50]: Addressed (see invalid_kafka_user_create.tf and its associated usage in acc_test/resource_kafka_user_test.go)

**Review by EngineersBox — DISMISSED** [review]
LGTM, pending reviews from Pete, Amrutha and Hendra that is

**Review by h3nd24 — DISMISSED** [review]

**Review by h3nd24 — DISMISSED** [review]

**Review by EngineersBox — APPROVED** [review]
LGTM

**Review by peter-toms — CHANGES_REQUESTED** [review]
Please add description to this PR

Please add doc details into https://github.com/instaclustr/terraform-provider-instaclustr/blob/master/docs/resources/kafka_user.md. otherwise looking good

**Review by h3nd24 — APPROVED** [review]

**Review by peter-toms — APPROVED** [review]

**Pull request merged**
